doc/releases/changelog-dev.md

```diff
@@ -4,7 +4,74 @@
 
 <h3>New features since last release</h3>
 
-* A tensor network templates module has been added. Quantum circuits with the shape of a matrix product state tensor network can now be easily implemented. Motivation and theory can be found in [arXiv:1803.11537](https://arxiv.org/abs/1803.11537). [(#1871)](https://github.com/PennyLaneAI/pennylane/pull/1871)
+* The `RotosolveOptimizer` has been generalized to arbitrary frequency spectra
+  in the cost function.
+  [(#2081)](https://github.com/PennyLaneAI/pennylane/pull/2081)
+
+  Previously, the `RotsolveOptimizer` was available for cost functions with
+  frequency spectra that only contained integers, and the maximal frequency
+  (instead of the number of frequencies) determined the cost of the optimization.
+  Now arbitrary frequencies are supported.
+
+  Consider the QNode
+  ```python
+  dev = qml.device("default.qubit", wires=2)
+
+  @qml.qnode(dev)
+  def qnode(x, Y):
+      qml.RX(2.5*x, wires=0)
+      qml.CNOT(wires=[0, 1])
+      qml.RZ(0.3*Y[0], wires=0)
+      qml.CRY(1.1*Y[1], wires=[1, 0])
+      return qml.expval(qml.PauliX(0)@qml.PauliZ(1))
+
+  x = np.array(0.8, requires_grad=True)
+  Y = np.array([-0.2, 1.5], requires_grad=True)
+  ```
+
+  Its frequency spectra can be easily obtained via `qml.fourier.qnode_spectrum`:
+  ```pycon
+  >>> spectra = qml.fourier.qnode_spectrum(qnode)(x, Y)
+  >>> spectra
+  {'x': {(): [-2.5, 0.0, 2.5]},
+   'Y': {(0,): [-0.3, 0.0, 0.3], (1,): [-1.1, -0.55, 0.0, 0.55, 1.1]}}
+  ```
+
+  We may then initialize the `RotosolveOptimizer` and minimize the QNode cost function
+  by providing this information about the frequency spectra. We also compare the cost at
+  each step to the initial cost.
+  ```pycon
+  >>> cost_init = qnode(x, Y)
+  >>> opt = qml.RotosolveOptimizer()
+  >>> for _ in range(2):
+  ...     x, Y = opt.step(qnode, x, Y, spectra=spectra)
+  ...     print(f"New cost: {np.round(qnode(x, Y), 3)}; Initial cost: {np.round(cost_init, 3)}")
+  New cost: 0.0; Initial cost: 0.706
+  New cost: -1.0; Initial cost: 0.706
+  ```
+
+  The optimization with `RotosolveOptimizer` is performed in substeps. The minimal cost
+  of these substeps can be retrieved by setting `full_output=True`.
+  ```pycon
+  >>> x = np.array(0.8, requires_grad=True)
+  >>> Y = np.array([-0.2, 1.5], requires_grad=True)
+  >>> cost_init = qnode(x, Y)
+  >>> opt = qml.RotosolveOptimizer()
+  >>> for _ in range(2):
+  ...     x, Y = opt.step(qnode, x, Y, spectra=spectra)
+  ...     print(f"New cost: {np.round(qnode(x, Y), 3)} reached via substeps {np.round(history, 3)}")
+  New cost: 0.0 reached via substeps [-0.  0.  0.]
+  New cost: -1.0 reached via substeps [-0.276 -0.276 -1.   ]
```

Comment:
maybe I am missing it, but where is `full_output=True` passed, and where is `history` defined?

Reply:
Oh, something must have gone wrong when inserting the code example. Thanks!